src/librustc_trans/type_.rs

```diff
@@ -268,6 +269,20 @@ impl Type {
         }
     }
 
+    pub fn fat_ptr_data_type(&self) -> Type {
```

Comment:
You'd want to call it on the pointee type, which you can easy get, by changing the second `&ty::TyRawPtr(..)` pattern to `&ty::TyRawPtr(mt)` and using `mt.ty`.